File: next-js-boilerplate/src/api/server/messages/decline-friend-request.ts
import { apiFetch } from "@/lib/api-client";
import { MESSAGES_FRIENDS_DECLINE_PREFIX } from "@/constants/api/urls";
import { POST } from "@/constants/api/methods";

export async function declineFriendRequestServer(
  userId: string,
): Promise<boolean> {
  const res = await apiFetch(MESSAGES_FRIENDS_DECLINE_PREFIX + userId, {
    method: POST,
  });
  return res.ok;
}

File: flutter-boilerplate/lib/api/server/messages/decline_friend_request.dart
import 'package:dio/dio.dart';
import 'package:flutter_boilerplate/lib/api_client.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../../constants/api/urls.dart';

final declineFriendRequestServerProvider = Provider(
  (ref) => DeclineFriendRequestServer(ref.read(dioProvider)),
);

class DeclineFriendRequestServer {
  final Dio _dio;

  DeclineFriendRequestServer(this._dio);

  Future<void> call(String requestId) async {
    await _dio.post<dynamic>('${Urls.declineFriendRequest}/$requestId');
  }
}
